src/content/lodash/string/kebabCase/vanilla.js

```diff
@@ -1,20 +1,53 @@
 const kebabCase = str =>
-  str
-    // Add hypen before capital letters followed by lowercase letters
-    .replace(/[A-Z][a-z]/g, match => `-${match}`)
-    // Everything that is not alphanumeric with an hypen
-    .replace(/[^A-Za-z0-9]/g, '-')
-    // Replace consecutive hyphens with a single hyphen
-    .replace(/-{2,}/g, '-')
-    // Remove hyphen from start and end of the string
-    .replace(/(^-|-$)/g, '')
-    .toLowerCase()
+    (String(str ?? "").match(/([0-9]+|([A-Z][a-z]+)|[a-z]+|([A-Z]+)(?![a-z]))/g) ?? [""])
```

Comment:
Given that this regex is way more complicated than the chunks in my previous version, we need some comments explaining what it's doing. Also the notes.md file should have a reference to the negative lookahead (as well as the `??`)

Reply:
I have added explanations in the `notes.md` file

English is not my strong point, so don't hesitate to correct me

Reply:
Do your best, the goal is to share and improve :)